### DISCUSSION TITLE
Cannot connect to socketio server in Hono + Deno environment

### DISCUSSION THREAD

**Discussion by preportteam** [socketio/socket.io#5199] [Q&A]
Hello everyone, I am getting this issue where my socketio client cannot connect to the server, it shows connection failed error in browser.
I am trying to setup the server in hono and deno, I specifically want to try and setup things through npm packages, which is possible in Deno now (through npm: specifiers), my hono instance works all well and response is given as expected, the problem arising is when I pass server to the socketio server instance, please refer code below : 

```
import { Server as HttpServer } from "node:http";
import { Hono } from "npm:hono";
import { Server } from "npm:socket.io";

const app = new Hono();
app.get("/", (c) => c.text("Hello Deno!"));
const server = Deno.serve({ port: 8080 }, app.fetch);

//Am trying to type cast or ignore types, also think the main issue is here only
const io = new Server(server as unknown as HttpServer, {
  cors: {
    origin: "*",
    credentials: true,
  },
});

io.use((socket, next) => {
  const handshake = socket.handshake.auth.token;
  if (handshake !== "abcd") {
    next(new Error("Not allowed"));
  } else {
    console.log("allowed");
    next();
  }
});

io.on("connection", (socket) => {
  console.log(`${socket.id} connected`);
  io.emit("hello", "world");
  socket.on("ping", (callback) => {
    callback();
  });
});

```

Everything works perfectly when I try to use expressjs, but here I can't get it working, am still new to socketio and environments setup, so help would be really appreciated, Thanks!

**Comment by preportteam** [comment]
> Hello everyone, I am getting this issue where my socketio client cannot connect to the server, it shows connection failed error in browser. I am trying to setup the server in hono and deno, I specifically want to try and setup things through npm packages, which is possible in Deno now (through npm: specifiers), my hono instance works all well and response is given as expected, the problem arising is when I pass server to the socketio server instance, please refer code below :
> 
> ```
> import { Server as HttpServer } from "node:http";
> import { Hono } from "npm:hono";
> import { Server } from "npm:socket.io";
> 
> const app = new Hono();
> app.get("/", (c) => c.text("Hello Deno!"));
> const server = Deno.serve({ port: 8080 }, app.fetch);
> 
> //Am trying to type cast or ignore types, also think the main issue is here only
> const io = new Server(server as unknown as HttpServer, {
>   cors: {
>     origin: "*",
>     credentials: true,
>   },
> });
> 
> io.use((socket, next) => {
>   const handshake = socket.handshake.auth.token;
>   if (handshake !== "abcd") {
>     next(new Error("Not allowed"));
>   } else {
>     console.log("allowed");
>     next();
>   }
> });
> 
> io.on("connection", (socket) => {
>   console.log(`${socket.id} connected`);
>   io.emit("hello", "world");
>   socket.on("ping", (callback) => {
>     callback();
>   });
> });
> ```
> 
> Everything works perfectly when I try to use expressjs, but here I can't get it working, am still new to socketio and environments setup, so help would be really appreciated, Thanks!

I think this issue would be regarding or related to socketio, so wanted to ask here

**Reply by calebeaires** [reply]
Hey, have you found a solution?
